lib/mixlibrary/core/shell/scripts/shelloutwrapper.rb

```diff
@@ -86,11 +86,23 @@ def shellout(eval_error)
               create_file()
       
               shellclass=Shell::ShellCall.new
+
+              Chef::Log::debug("Script Contents:")
+              Chef::Log::debug("-----------------------------------------------------")
+              Chef::Log::debug(@code)
+              Chef::Log::debug("-----------------------------------------------------")
+
               if(eval_error)  
                 result = shellclass.shell!("#{@exe} #{@flags} #{@script_file.path}", @shellout_options)
+                Chef::Log.debug("Command output:#{result.stdout}")
+                Chef::Log.debug("Command Error:#{result.stderr}")
+                Chef::Log.debug("Command Exit Code:#{result.exitstatus}")
                 return result  
               else
                 result = shellclass.shell("#{@exe} #{@flags} #{@script_file.path}", @shellout_options)
+                Chef::Log.debug("Command output:#{result.stdout}")
+                Chef::Log.debug("Command Error:#{result.stderr}")
+                Chef::Log.debug("Command Exit Code:#{result.exitstatus}")
                 return result
               end
```

Comment:
Would it make sense to move:

    Chef::Log.debug("Command output:#{result.stdout}")
    Chef::Log.debug("Command Error:#{result.stderr}")
    Chef::Log.debug("Command Exit Code:#{result.exitstatus}")
    return result

to outside of the "if" statement? Those lines appear to be identical in both the "if" and "else" clauses.